<launch>
    <!-- 启动 USB 相机 -->
    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" launch-prefix="bash -c 'rosrun usb_cam usb_cam_node _video_device:=/dev/video0 _pixel_format:=mjpeg _image_width:=1280 _image_height:=480 2>/dev/null; $0 $@'" />

    <!-- 图像左右分割节点 -->
    <node pkg="stereo_splitter" type="split_stereo_image.py" name="split_stereo_image" output="screen" />

    <!-- 执行 YOLOv8 目标检测（使用 LD_PRELOAD） -->
    <!-- 1:shpaes, 2:holes, 3:balls-->

    <node pkg="yolo_bridge" type="yolo_wrapper.sh" name="yolov8_tensorrt" output="screen" launch-prefix="bash" >
        <param name="detect_mode" value="3" />
        <param name="top_k" value="3" />
        <param name="visualization" value="False" />
    </node>

</launch>

